controller/src/main/java/io/pravega/controller/server/Main.java

```diff
@@ -86,8 +88,30 @@ public static void main(String[] args) {
                     .restServerConfig(Optional.of(restServerConfig))
                     .build();
 
+            Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
+                log.error("Thread {} with stackTrace {} failed with uncaught exception", t.getName(), t.getStackTrace(), e);              
```

Comment:
Is this an error or warn message? It is an error in the case the thread dying is an unrecoverable situation.

Reply:
We are registering callback for `uncaughtException` which essentially means thread faced an erroneous case with an exception that was unhandled.. hence error level logging.